design/src/rfcs/rfc0006_crates_io_publishing.md

```diff
@@ -0,0 +1,269 @@
+makes sense to apply these to `patch` revisions as much as possible. That way, if a model change was made only
+to S3, then only the S3 crate would be changed rather than every AWS SDK crate, of which there are more than 260.
+
+AWS SDK crates will follow this process for version bumps:
+
+1. Bump the `minor` version in tandem with _all_ other SDK crates and `aws-config` if the smithy-rs version used to
+   generate the SDK changed. This ensures changes to runtime crates and/or code generation logic will result in
+   a `minor` version bump.
+2. Bump the `patch` version if the smithy-rs version remained the same _AND_ if there are code changes to
+   the SDK crate. This allows backwards compatible model changes to apply as patch revisions.
+
+The version bump requirements for runtime crates are flexible and deferred to the short-term and long-term proposals.
+
+### Yanking
+
+Mistakes will inevitably be made, and a mechanism is needed to yank packages while keeping the latest version
+of the SDK successfully consumable from crates.io. To keep this simple, the entire published batch of crates
+will be yanked if any crate in that batch needs to be yanked. For example, if 260 crates were published in a batch,
+and it turns out there's a problem that requires yanking one of them, then all 260 will be yanked. Attempting to do
+partial yanking will require a lot of effort and be difficult to get right. Yanking should be a last resort.
+
+Concrete Scenarios
+------------------
+
+The following are example scenarios that may occur that the publishing implementation must handle.
+
+1. **AWS models were updated**: This results in `patch` version bumps for all AWS SDK crates _that had changes_.
+2. **Bug fixed in runtime crate**: Runtime crate is `minor` version bumped. All runtime crates that depend on it are
+   also `minor` version bumped, and all of the AWS SDK crates are `minor` version bumped.
+3. **Bug fixed in codegen**: All AWS SDK crates are `minor` version bumped.
+4. **CVE discovered in external dependency**: If the external dependency is fixed in a `patch` revision where it will
+   automatically upgrade via semver, then no action is required. Otherwise, update the dependency where it is
+   included, and then follow the normal version bump process from there.
+5. **CVE discovered in runtime crate**: If it can be fixed in a backwards compatible way, then `patch` revision the
+   runtime crate with the fix. Otherwise, `minor` version bump the runtime crate and follow the normal process from there.
+6. **CVE discovered in generated code**: Same process as a bug fix in codegen.
+7. **Accidental release of breaking change**: Run publish process in reverse, yanking the published packages.
+8. **Buggy update to runtime crate**: If possible, roll forward with a `patch` revision. Otherwise, run publish process
+   in reverse yanking the published packages.
+
+Short-term Proposal
```

Comment:
a section with the estimated effort in dev weeks for the short term and long term proposal would probably be helpful